robocop/checkers/tags.py

```diff
@@ -42,4 +48,73 @@ def check_tag(self, tag, node):
         if ' ' in tag.value:
             self.report("tag-with-space", node=node, lineno=tag.lineno, col=tag.col_offset + 1)
         if 'OR' in tag.value or 'AND' in tag.value:
+            self.report("tag-with-or-and", node=node, lineno=tag.lineno, col=tag.col_offset + 1)
+        if tag.value.startswith('robot:') and tag.value != 'robot:no-dry-run':
             self.report("tag-with-reserved", node=node, lineno=tag.lineno, col=tag.col_offset + 1)
+
+
+class TagScopeChecker(VisitorChecker):  # TODO: load tags also from __init__.robot
+    """ Checker for tag scopes. If the all tests in suite have the same tags it will suggest using Force Tags """
+    msgs = {
+        "0605": (
+            "could-be-forced-tags",
+            'All tests in suite share those tags: "%s". You can define them in Force Tags in suite settings instead',
+            MessageSeverity.INFO
+        ),
+        "0606": (
+            "tag-already-set-in-force-tags",
+            "This tag is already set by Force Tags in suite settings",
+            MessageSeverity.INFO
+        ),
+        "0607": (
+            "unnecessary-default-tags",
+            "Tags defined in Default Tags are always overwritten",
+            MessageSeverity.INFO
+        )
+    }
+
+    def __init__(self, *args):
+        self.tags = []
+        self.force_tags = []
+        self.default_tags = []
+        self.force_tags_node = None
+        self.default_tags_node = None
+        self.test_cases_count = 0
+        super().__init__(*args)
+
+    def visit_File(self, node):  # noqa
+        self.tags = []
+        self.force_tags = []
+        self.test_cases_count = 0
+        self.force_tags_node = None
+        super().visit_File(node)
```

Comment:
thanks to overriding visit_File in this way, we make sure whole tests is pared and all nodes are hit before we start our analyze. This approach can be reused in different checkers if we need to compare values (like check for duplicates)